pwiz_tools/Skyline/FileUI/PeptideSearch/DDASearchControl.cs

```diff
@@ -74,11 +73,30 @@ protected virtual void UpdateTaskbarProgress(TaskbarProgress.TaskbarStates state
 
         private void UpdateSearchEngineProgress(IProgressStatus status)
         {
-            var newEntry = new ProgressEntry(DateTime.Now, status.Message);
+            string message = status.IsError ? status.ErrorException.ToString() : status.Message;
+
+            var newEntry = new ProgressEntry(DateTime.Now, message);
             _progressTextItems.Add(newEntry);
             txtSearchProgress.AppendText($@"{newEntry.ToString(showTimestampsCheckbox.Checked)}{Environment.NewLine}");
-            UpdateTaskbarProgress(TaskbarProgress.TaskbarStates.Normal, status.PercentComplete);
-            progressBar.Value = status.PercentComplete;
+
+            if (status.IsError)
+            {
+                Program.ReportException(status.ErrorException);
```

Comment:
You should use "MessageDlg.ShowWithException" here instead.

Program.ReportException is only for things that are actual bugs where we want the user to submit a bug report to the exception web site.

Reply:
So `ShowWithException` doesn't allow the user to report the error with a button press? DDA support is still pretty new: I'm thinking that for the near future we should allow users to report these exceptions easily. Perhaps we can just add catches for the common non-bug errors that get reported here, and handle those with `ShowWithException`, but let unexpected errors fall through to `ReportException`?

Reply:
You certainly could show a message box with multiple buttons on it, telling them to push one of the buttons if they think the error was a bug (and pushing that button would call Program.ReportException to bring up the ReportErrorDlg). There is no place where we currently do that in the code, but it would be a reasonable thing to do.

You would not want the user to see the ReportErrorDlg without that other message box first. The information on the ReportErrorDlg is presented in a way that is very hard for users to read, so even when the error message is straightforward (e.g. "The file already exists"), users will typically not be able to figure out what went wrong.